specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2021-03-01-preview/appplatform.json

```diff
@@ -384,6 +497,126 @@
         }
       },
       "x-ms-external": true
+    },
+    "BuildResource": {
+      "description": "Build resource payload",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BuildResourceProperties",
+          "description": "Properties of the build resource",
+          "x-ms-client-flatten": false
+        }
+      }
+    },
+    "BuildResourceProperties": {
+      "description": "Build resource properties payload",
+      "type": "object",
+      "properties": {
+        "Builder":{
+          "description": "The name of builder to build the source code",
+          "type": "string"
+        },
+        "Name":{
+          "description": "The name of this build",
+          "type": "string"
+        },
+        "RelativePath":{
+          "description": "The relative path of source code",
+          "type": "string"
+        },
+        "Env": {
+          "description": "The environment variables for this build",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "BuildResult": {
```

Comment:
Why not reuse `BuildResultResource` but create a `BuildResult` as the wrapper of id of build result ?